python/lsst/daf/butler/datastores/fileLikeDatastore.py

```diff
@@ -899,8 +899,29 @@ def get(self, ref: DatasetRef, parameters: Optional[Mapping[str, Any]] = None) -
         allGetInfo = self._prepare_for_get(ref, parameters)
         refComponent = ref.datasetType.component()
 
+        # Create mapping from component name to related info
+        allComponents = {i.component: i for i in allGetInfo}
+
+        # By definition the dataset is disassembled if we have more
+        # than one record for it.
         isDisassembled = len(allGetInfo) > 1
 
+        # Look for the special case where we are disassembled but the
+        # component is a read-only component that was not written during
+        # disassembly. For this scenario we need to check that the
+        # component requested is listed as a read-only component for the
+        # composite storage class
+        isDisassembledReadOnlyComponent = False
+        if isDisassembled and refComponent:
+            # The composite storage class should be accessible through
+            # the component dataset type
+            compositeStorageClass = ref.datasetType.parentStorageClass
+
+            # Assume something else will handle things if we can not find
+            # it
```

Comment:
Is this actually a reasonable assumption?  Is there some other part of the system that could handle this case?

Reply:
If the parent storage class is not available this will assume it's a real component being requested. At that point the branch below that looks up the disassembled component will fail and you'll get an error about the component not being recognized.